blockchain/chain.go

```diff
@@ -2275,8 +2274,13 @@ func New(ctx context.Context, config *Config) (*BlockChain, error) {
 		"%d, block index: %d, spend journal: %d", b.dbInfo.version,
 		b.dbInfo.compVer, b.dbInfo.bidxVer, b.dbInfo.stxoVer)
 
+	// Fetch and log the UTXO backend versioning info.
+	utxoDbInfo, err := b.utxoCache.FetchBackendInfo()
```

Comment:
This seems a bit odd to me to fetch database details from the backend via the cache.  Perhaps it's just a temporary thing that will be removed in later commits, but it seems like this is something that would either:

a. Be internal to the cache and probably logged by it when initialized; or
b. The cache would be expected to take an already initialized backend in which case logging it outside of the cache would make sense

I think I prefer option (b) because the overall design here is such that the cache is designed to sit on top of an arbitrary backend.  Thus, it seems to me like it would be a caller responsibility to initialize the backend create the cache with it. What do you think?

Reply:
I agree and went back and forth a bit on this.  I updated this to the second option so that `BlockChain` now holds a `UtxoBackend` reference and is responsible for initializing it.

I originally went down this path since I was trying to avoid having anything other than the `UtxoCache` interacting with the `UtxoBackend` directly since the `UtxoBackend` really shouldn't be queried directly, and any queries or updates should be routed through the `UtxoCache`.  However, since `BlockChain` is initializing the `UtxoBackend`, I think it makes sense for `BlockChain` to have a reference to it just for initialization purposes.

Reply:
Right.  I definitely see the appeal in trying to limit the potential misuse of it by not even exposing it directly, and that did occur to me as well, which is why I had listed option (a) as well.  I believe you can still do that without issue.  Looking at all of the most recent updates, I see that the backend is only ever used outside of the cache in the initialization path.  Since that is the case, I don't see why you need a reference to it on `Blockchain` at all.  It is only used in `New` (which already has access to it) and `initChainState` which is called from `New` and thus could take it as a parameter.

Reply:
Good call, I like that idea.  I pushed an update for this as suggested.